### PR TITLE
AC-316 - Remove code duplicates

### PR DESCRIPTION
**Pull request by tmarzeion**
**TODO**:
- [x] `PatientDashboardVisitsPresenter`
- [x] `VisitApi`
- [x] `FormDisplayPageFragment`
- [x] `ActiveVisitPresenter`
- [x] `FilterUtil`
- [x] `AddEditPatientFragment`
- [x] `LastViewedPatientsPresenter`
- [ ] ~~`SimilarPatientsRecyclerViewAdapter`~~*
- [ ] ~~`MergePatientsRecycleViewAdapter`~~*
- [ ] ~~`SyncedPatientsRecyclerViewAdapter`~~*
- [ ] ~~`LastViewedPatientRecyclerViewAdapter`~~*

 :tada:  **DONE** :tada: 
________________________________

 `*` - Won't fix, since extracting common code from adapter would make it less readable.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/utilities/FilterUtil.java [hunk 1]
```diff
@@ -30,25 +32,72 @@
...
+
         }
         return filteredList;
     }
 
+    public static List<Visit> getPatientsWithActiveVisitsFilteredByQuery(List<Visit> visitList, String query) {
+        List<Visit> filteredList = new ArrayList<>();
+
+        for (Visit visit : visitList) {
+            Patient patient = new PatientDAO().findPatientByID(visit.getPatient().getId().toString());
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/utilities/FilterUtil.java]: I think we shouldn't call DAO here ?

Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/utilities/FilterUtil.java]: `Patient patient = visit.getPatient();` looks much better :man_facepalming:

Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/utilities/FilterUtil.java]: @AdamGrzybkowski Fixed :)

File: openmrs-client/src/main/java/org/openmrs/mobile/listeners/retrofit/DownloadMultiplePatientsCallbackListener.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package org.openmrs.mobile.listeners.retrofit;
+
+import org.openmrs.mobile.models.Patient;
+
+import java.util.List;
+
+public interface DownloadMultiplePatientsCallbackListener extends DefaultResponseCallbackListener {
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/listeners/retrofit/DownloadMultiplePatientsCallbackListener.java]: @tmarzeion Do we need this ?

**Comment by AdamGrzybkowski** [comment]
@tmarzeion please see my comment above and resolve conflicts :)

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/lastviewedpatients/LastViewedPatientsPresenter.java [hunk 3]
```diff
@@ -31,7 +31,7 @@
 import retrofit2.Callback;
 import retrofit2.Response;
 
-public class LastViewedPatientsPresenter implements LastViewedPatientsContract.Presenter{
+public class    LastViewedPatientsPresenter implements LastViewedPatientsContract.Presenter{
```
Comment (shivtej1505) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/lastviewedpatients/LastViewedPatientsPresenter.java]: @tmarzeion , unnecessary spaces here.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/lastviewedpatients/LastViewedPatientsPresenter.java [hunk 4]
```diff
@@ -64,95 +64,94 @@ public void start() {
...
+    private void setViewBeforePatientDownload() {
+        if (!mLastViewedPatientsView.isRefreshing()) {
+            mLastViewedPatientsView.setSpinnerVisibility(true);
+        }
+        mLastViewedPatientsView.enableSwipeRefresh(false);
+        mLastViewedPatientsView.setEmptyListVisibility(false);
+        mLastViewedPatientsView.setListVisibility(false);
+    }
+
+    private void setViewAfterPatientDownloadErrored(String errorMessage) {
```
Comment (shivtej1505) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/lastviewedpatients/LastViewedPatientsPresenter.java]: @tmarzeion , I feel `setViewAfterPatientDownloadError` would be better function name.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/lastviewedpatients/LastViewedPatientsPresenter.java (hunk 3 above)
Comment (shivtej1505) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/lastviewedpatients/LastViewedPatientsPresenter.java]: @tmarzeion , unnecessary spaces here.

File: openmrs-client/src/main/java/org/openmrs/mobile/listeners/retrofit/DownloadMultiplePatientsCallbackListener.java (hunk 2 above)
Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/listeners/retrofit/DownloadMultiplePatientsCallbackListener.java]: @AdamGrzybkowski Sorry, it was leftover which was useful before I realized my idea was kind of retarded :) Fixed

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/lastviewedpatients/LastViewedPatientsPresenter.java (hunk 4 above)
Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/lastviewedpatients/LastViewedPatientsPresenter.java]: @shivtej1505 Fixed :) thx for feedback

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/lastviewedpatients/LastViewedPatientsPresenter.java (hunk 3 above)
Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/lastviewedpatients/LastViewedPatientsPresenter.java]: @shivtej1505 Fixed :)

**Pull request merged**